class.js

```diff
@@ -143,19 +159,14 @@ class Pokemon {
   }
 
   evolve(){
-    //Basic Pokemon: Pikachu, Charmander, Squirtle, Bulbasaur, Pidgey, Caterpie, Ekans, Oddish, Doduo, Vulpix, Jigglypuff, Zubat
-    //2nd level Pokemon: Raichu, Charmeleon, Wartortle, Ivysaur, Pidgeotto, Metapod, Arbok, Gloom, Dodrio, Ninetales, Wigglytuff, Golbat
-    //3rd level: Charizard, Blastoise, Venusaur, Pigeot, Butterfree, Vileplume, Crobat
-    //2nd level unevolvable: Raichu, Arbok, Dodrio, Ninetales, Wigglytuff
-    
-    //Actually when it's already 3rd level it shouldn't try to evolve - so we just have to check for the other 5. EZ
-    if (this.name == "Raichu" || this.name == "Arbok" || this.name == "Dodrio" || this.name == "Ninetales" || this.name == "Wigglytuff" || this.name == "Clefable"){
+    if (this.name == "Raichu" || this.name == "Arbok" || this.name == "Dodrio" || this.name == "Ninetales" || this.name == "Wigglytuff" || this.name == "Clefable" || this.name == "Lapras" || this.name == "Raticate" || this.name == "Golduck"){
       console.log("Evolution failed");
       //fucking don't do anything I guess
     }
     else {
-      var newNameIndex = pkmList.indexOf(this.name) + 1;
-      this.name = pkmList[newNameIndex];
+      var newNameIndex = getPokemonId(this.name) + 1;
+      this.name = pkmList[newNameIndex][0];
```

Comment:
You might want to make the elements of `pkmList` objects with `name` and `type` properties rather than having to remember what value is in which position in an array.